packages/node_modules/brightspace-auth-keys-dynamodb-store/test/public-key-store.spec.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+const
+	DynamoPublicKeyStore = require('../src/public-key-store'),
+	sinon = require('sinon'),
+	chai = require('chai'),
+	assert = require('chai').assert,
+	DynamoDbLocal = require('dynamodb-local'),
+	AWS = require('aws-sdk'),
+	chaiAsPromised = require('chai-as-promised'),
+	clock = require('../src/clock');
+
+chai.use(chaiAsPromised);
+
+describe('public-key-storeintegration tests', function() {
+	const expectedPublicKey = JSON.stringify({
+		kid: 'TestPublicKey'
+	});
+
+	const dynamoLocalPort = 8000;
+	const expiryTime = clock() + 10000;
+	const oldExpiryTime = clock() - 10000;
+	const tableName = 'auth_public_keys';
+
+	let provisioner;
+	let db;
+
+	before(() => {
+		return DynamoDbLocal.launch(dynamoLocalPort, null, ['-sharedDb']).then(() => {
+			db = new AWS.DynamoDB({
+				endpoint: new AWS.Endpoint('http://localhost:8000'),
+				accessKeyId: 'test',
+				secretAccessKey: 'test',
+				region: 'test'
+			});
+			return DynamoPublicKeyStore.createTable(db, tableName).then(() => {
+				provisioner = new DynamoPublicKeyStore(
+					db,
+					tableName
+				);
+			});
+		});
+	});
+
+	after(() => {
+		DynamoDbLocal.stop(dynamoLocalPort);
+	});
+
+	it('calling create table does nothing if it already exists', () => {
+		return DynamoPublicKeyStore.createTable(db, tableName).then(() => {
+			return DynamoPublicKeyStore.createTable(db, tableName);
+		});
+	});
+
+	it('stores the public key into the AUTH_PUBLIC_KEYS table', () => {
+		return provisioner._storePublicKey(expectedPublicKey, expiryTime)
+			.then(() => {
+				return provisioner._lookupPublicKeys();
+			})
+			.then(retrievedKeys => {
+				// Rather than checking for equality, assert that expectedPublicKey is in the returned array.
+				// The contents of the array are not guaranteed, and may contain actual keys. The same applies for the test below.
```

Comment:
Probably better to just have the comment again rather than positionally referring to another test :)

Reply:
Just realized I did it first ;)